File: src/com/woezelmann/showas/EditorWrapper.java
package com.woezelmann.showas;

import com.intellij.lang.Language;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import com.intellij.ui.EditorTextField;

import javax.swing.*;
import java.awt.*;

/**
 * @author wozniak
 * @since 16.05.12
 */
public class EditorWrapper extends DialogWrapper {

    private JPanel panel = new JPanel(new BorderLayout());

    protected EditorWrapper(Language language, Project project, String text) {
        super(project);
        super.setOKActionEnabled(false);

        init();

        EditorTextField editorTextField = new ScrollableLanguageTextField(language, project, text);
        editorTextField.setOneLineMode(false);
        editorTextField.setPreferredSize(new Dimension(800, 600));

        editorTextField.setVisible(true);

        panel.add(editorTextField);
    }

    @Override
    protected JComponent createCenterPanel() {
        return panel;
    }
}

File: src/main/kotlin/com/woezelmann/showas/EditorWrapper.kt
package com.woezelmann.showas

import com.intellij.ui.EditorTextField
import java.awt.BorderLayout
import java.awt.Dimension
import javax.swing.JComponent
import javax.swing.JPanel

import com.intellij.lang.Language;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;


class EditorWrapper : DialogWrapper {
    private val panel = JPanel(BorderLayout())

    constructor(language: Language, project: Project, text: String) : super(project){
        super.setOKActionEnabled(false)
        init()
        val editorTextField: EditorTextField = ScrollableLanguageTextField(language, project, text)
        editorTextField.setOneLineMode(false)
        editorTextField.preferredSize = Dimension(800, 600)
        editorTextField.isVisible = true
        panel.add(editorTextField)
    }

    override fun createCenterPanel(): JComponent {
        return panel
    }
}